tools/yarn_install.R

```diff
@@ -1,9 +1,129 @@
+library(sass)
+
 if (Sys.which("yarn") == "") {
   stop("The yarn CLI must be installed and in your PATH")
 }
 
 withr::with_dir("inst", system("yarn install"))
 
+# ----------------------------------------------------------------------
+# Add known vendor prefixes to bootstrap.scss (since we don't want
+# to rely on a node run-time to prefix after compilation)
+# https://github.com/twbs/bootstrap/blob/d438f3/package.json#L30
+# ----------------------------------------------------------------------
+
+scss_files <- dir(
+  "inst/node_modules/bootstrap/scss",
+  recursive = TRUE, full.names = TRUE
+)
+scss_src <- lapply(scss_files, readLines)
+
+add_prefixes <- function(src, property, ok_values = NULL) {
+  pattern <- paste0("^\\s*", property, ":\\s*(.+);")
+  idx <- grep(pattern, src)
+  for (i in idx) {
+    prop <- src[[i]]
+    if (length(ok_values)) {
+      value <- regmatches(prop, regexec(pattern, prop))[[1]][2]
+      vals <- strsplit(value, "\\s+")[[1]]
+      if (all(grepl(paste(ok_values, collapse = "|"), vals))) next
+    }
+    leading_ws <- regmatches(prop, regexpr("^\\s+", prop))
+    prop_prefixes <- paste0(
+      leading_ws,
+      c("-webkit-", "-moz-", "-ms-"),
+      sub("^\\s+", "", prop),
+      collapse = "\n"
+    )
+    src[[i]] <- paste0(prop, "\n", prop_prefixes)
+  }
+  src
+}
+
+# Unconditionally prefix the following CSS properties
+needs_prefix <- c(
+  "appearance", "user-select", "backdrop-filter",
+  "backface-visibility", "touch-action"
+)
+for (prop in needs_prefix) {
+  scss_src <- lapply(scss_src, add_prefixes, prop)
+}
+# Conditionally prefix text-decoration if its not CSS2 compliant
+# https://www.w3.org/TR/CSS2/text.html#lining-striking-props
+# https://developer.mozilla.org/en-US/docs/Web/CSS/text-decoration
+# https://caniuse.com/#search=text-decoration
+scss_src <- lapply(scss_src, add_prefixes, "text-decoration", c("none", "underline", "overline", "line-through", "blink", "inherit"))
+
+# Write modified source to disk
+invisible(Map(writeLines, scss_src, scss_files))
+
+# ----------------------------------------------------------------------
+# Check to make sure we're not missing any vendor prefixes
+# that we don't already know about in the distributed CSS
+# ----------------------------------------------------------------------
+
+find_prefixed_css <- function(css) {
+  vendors <- c("webkit", "moz", "ms")
+  prefixes <- lapply(vendors, function(vendor) {
+    pattern <- sprintf("-%s-([a-z-]+)", vendor)
```

Comment:
```suggestion
    pattern <- sprintf("-%s-([^:]+)", vendor)
```

Reply:
There are a few cases where this won't work 

https://github.com/twbs/bootstrap/blob/af4e37631/dist/css/bootstrap.css#L1632
https://github.com/twbs/bootstrap/blob/af4e37631c526061d853881f11c5363f61907ba5/dist/css/bootstrap.css#L5575
https://github.com/twbs/bootstrap/blob/af4e37631/dist/css/bootstrap.css#L5767

Reply:
How about we compromise on the following?

```r
pattern <- sprintf("-%s-([^:|;| |\\)]+)", vendor)
```